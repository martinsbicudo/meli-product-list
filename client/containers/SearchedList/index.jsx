import React from 'react';
import { string } from 'prop-types';

const SearchedList = ({
  search,
}) => (
  <span>{search}</span>
);

SearchedList.propTypes = {
  search: string.isRequired,
};

export default SearchedList;
