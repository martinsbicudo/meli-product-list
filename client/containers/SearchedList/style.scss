.SearchedList {
  padding: 2rem 0;
  width: 100%;
  display: flex;
  flex-direction: column;

  &Title {
    font-size: 2.2rem;
    margin-top: 1rem;
    margin-bottom: 3rem;
    width: 100%;

    @include query-max("tablet") { text-align: center; }
  }

  &Box {
    display: flex;
    align-items: stretch;

    @include query-max("mobile") { flex-direction: column; }

    @include query-min("mobile") { flex-wrap: wrap; }
  }

  &Item {
    cursor: pointer;
    width: 100%;

    @include query-min("desktop") { width: calc((100% / 3) - 1rem); }

    @include query-between("tablet", "desktop") { width: calc(50% - 1rem); }

    &:not(:last-child) { margin-bottom: 2rem; }
  }

  &Item:not(:nth-child(3n)) {
    @include query-min("desktop") {
      margin-right: 1.5rem;
    }
  }

  &Item:nth-child(odd) {
    @include query-between("tablet", "desktop") {
      margin-right: 2rem;
    }
  }


  &Query {
    display: block;
    white-space: nowrap;
    width: 100%;
    overflow: hidden;
    text-overflow: ellipsis;
  }
}
