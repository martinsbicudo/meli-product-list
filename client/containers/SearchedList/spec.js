import SearchedList from '.';
import { useLoadingStatus } from '@Hooks';
import { searchProducts } from '@Actions';

// MOCK USE-LOADING-STATUS
const initialState = [{
  loading: false,
  status: {
    success: false,
    error: false,
  },
}, {
  setLoading: jest.fn(),
  setStatus: jest.fn(),
}];
jest.mock('../../hooks', () => ({
  useLoadingStatus: jest.fn(),
}));
// /

// MOCK USE-LOADING-STATUS
const testMessageTitle = 'TEST_TITLE';
const testMessagePrice = '300';
const testMessageImage = 'TEST_IMAGE';
const product = () => ({
  id: Math.random(),
  title: testMessageTitle,
  price: testMessagePrice,
  thumbnail: testMessageImage,
});
const initialProducts = ({
  resolve,
  reject,
  finish,
}) => {
  resolve({
    data: [
      product(),
      product(),
      product(),
      product(),
      product(),
    ],
  });
  reject();
  finish();
};
jest.mock('../../actions', () => ({
  searchProducts: jest.fn(),
}));
// /

describe('Containers - <SearchedList />', () => {
  const search = 'TEST_SEARCH';
  const createWrapper = () => render(
    <SearchedList search={search} />,
  );

  test('Should finds 5 LoadingBar', () => {
    useLoadingStatus.mockImplementation(() => initialState);
    const { container } = createWrapper();

    expect(container.querySelectorAll('.LoadingBar')).toHaveLength(5);
  });

  test('Should finds EmptyAlert', () => {
    useLoadingStatus.mockImplementation(() => ([{
      ...initialState[0],
      status: {
        success: true,
        error: false,
      },
    }, { ...initialState[1] }]));
    const { getByText } = createWrapper();

    expect(getByText('Tente pesquisar um produto diferente :)')).toBeInTheDocument();
  });

  test('Should finds 5 items', () => {
    useLoadingStatus.mockImplementation(() => ([{
      ...initialState[0],
      status: {
        success: true,
        error: false,
      },
    }, { ...initialState[1] }]));
    searchProducts.mockImplementation(initialProducts);
    const { container } = createWrapper();

    expect(container.querySelectorAll('[alt="TEST_TITLE"]')).toHaveLength(5);
  });

  test('Wrapper Snapshot', () => {
    useLoadingStatus.mockImplementation(() => initialState);
    const { asFragment } = createWrapper();

    expect(asFragment()).toMatchSnapshot();
  });
});
