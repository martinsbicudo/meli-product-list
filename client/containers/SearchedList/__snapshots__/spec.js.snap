// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Containers - <SearchedList /> Wrapper Snapshot 1`] = `
<DocumentFragment>
  <section
    class="SearchedList"
  >
    <div
      class="Loading"
    >
      <div
        class="LoadingBars"
      >
        <i
          class="LoadingBar"
        />
        <i
          class="LoadingBar"
        />
        <i
          class="LoadingBar"
        />
        <i
          class="LoadingBar"
        />
        <i
          class="LoadingBar"
        />
      </div>
    </div>
  </section>
</DocumentFragment>
`;
